### PR TITLE
[wasm] Skip tests for ExpectContinue HTTP header

### LINKED ISSUE #53876: [wasm] HttpRequestMessage.Headers.ExpectContinue is ignored on browser
In unit test `System.Net.Http.Functional.Tests.HttpClientHandler_RemoteServerTest.PostAsync_ExpectContinue_Success(expectContinue: True, version: 1.1)` on WASM in Browser

```
Assert.Contains() Failure Not found: "Expect": "100-continue"
   at System.Net.Http.Functional.Tests.HttpClientHandler_RemoteServerTest.PostAsync_ExpectContinue_Success(Nullable`1 expectContinue, Version version)
```

### PR DESCRIPTION
**Pull request by pavelsavara**
[Expect:](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Expect
) `100-continue` header is not supported by any of the normal browsers.

See
[Gecko issue](https://bugzilla.mozilla.org/show_bug.cgi?id=803673)
[Chromium issue](https://bugs.chromium.org/p/chromium/issues/detail?id=174906) 

Skip related tests fixes https://github.com/dotnet/runtime/issues/53876

### CODE REVIEW

File: src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpRequestHeaders.cs [hunk 1]
```diff
@@ -19,8 +19,11 @@ public sealed class HttpRequestHeaders : HttpHeaders
 
         private object[]? _specialCollectionsSlots;
         private HttpGeneralHeaders? _generalHeaders;
+
+#if !TARGET_BROWSER
```
Comment (stephentoub) [inline src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpRequestHeaders.cs:23]: Please revert the changes in this file.  If we want to throw a PNSE from the browser handler, we can do that in the browser handler in response to an Expect: 100-continue header being part of the request, or just let it fail or be ignored or whatever the browser will actually do with the header.

Comment (pavelsavara) [reply src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpRequestHeaders.cs]: Okay will let the browser deal with it, which is that they ignore it silently. I didn't have strong preference either way.

**Review by ManickaP — APPROVED** [review]
LGTM, thanks.

**Review by stephentoub — APPROVED** [review]
